src/BinOp.chpl

```diff
@@ -946,29 +964,31 @@ module BinOp
       }
       var repMsg = "created %s".format(st.attrib(rname));
       return new MsgTuple(repMsg, MsgType.NORMAL);
-    } else if (val.type == int && r.etype == uint) {
+    }
+    else if (e.etype == int && val.type == uint) ||
+            (e.etype == uint && val.type == int) {
```

Comment:
Can you explain why these cases no longer rely on the `r.etype`.

Reply:
yah! basically i was updating it to be consistent with the vector scalar case